launcher/launcher.rb

```diff
@@ -0,0 +1,147 @@
+#!/usr/bin/env ruby
+CREW = `dirname $(which crew)`
+
+# use chromebrew variable
+require "#{CREW}/../lib/crew/lib/const"
+require "#{CREW}/../lib/crew/lib/color"
+require 'em-websocket'
+require 'fileutils'
+
+PID = CREW_PREFIX + '/tmp/server.pid'
+LOG = CREW_PREFIX + '/tmp/server.log'
+PWA_PREFIX = CREW_PREFIX + '/lib/pwa'
+PORT = 25500
+PATH = "#{CREW_PREFIX}/share/pixmaps " + "/usr/local/share/icons " + "#{Dir.home}/.local/ " + PWA_PREFIX
+
+HELP = <<EOF
+launcher: Add GUI applications shortcut to Chrome/Chromium OS launcher
+
+Usage:
+  stop            stop the launcher server if running
+  start           start the launcher server for shortcut
+  help            show this message
+  new <command>   make a new shortcut
+
+EOF
+
+JS = <<EOF
+chrome.app.runtime.onLaunched.addListener(function () {
+    const ws = new WebSocket('ws://0.0.0.0:#{PORT}');
+    ws.onopen = function () { ws.send('CMD'); }
+});
+EOF
+
+JSON = <<EOF
+{
+  "manifest_version": 2,
+  "app": {
+    "background": {
+      "scripts": [ "background.js" ]
+    }
+  },
+  "name": "NAME",
+  "version": "1.0",
+  "description": "INFO",
+  "icons": { "1024": "icon.png" },
+  "minimum_chrome_version": "56.0"
+}
+EOF
+
+POST_INSTALL = "
+You're almost done, three more steps: 
+
+Go to chrome://extensions in Chrome
+Ensure that the 'Developer mode' checkbox in the top right-hand corner is checked.
+Click on Load Unpacked and select the '<CMD>.launcher' folder.".lightblue
+
+# kill pid of the previous server in the log file (if exist)
+def stop_server
+  begin
+    Process.kill 'KILL', File.read(PID).to_i if File.exist?(PID)
+  rescue
+  end
+end
+
+
+def desktop_icon_name_finder (app)
+  # find out the info, name and icon of the command
+    
+  @desktop = `find / -name #{app}.desktop 2> /dev/null | sed -e '/stateful_partition/d' | tail -n1`.tr("\n", '')
```

Comment:
Is there a consistent pattern for where `#{app}.desktop` is located?  If so, why would we need to `find`?